dynamorm/model.py

```diff
@@ -459,3 +459,13 @@ def update(self, conditions=None, update_item_kwargs=None, **kwargs):
             setattr(self, key, val)
 
         return resp
+
+    def delete_item(self):
```

Comment:
I would like this to be named `delete`, to match the semantics of `save` and `update`.

The table method can remain delete_item on the Table class.